src/library/clementine/clementinedbconnection.cpp

```diff
@@ -0,0 +1,254 @@
+#include "library/clementine/clementinedbconnection.h"
+
+#include <QFile>
+#include <QFileInfo>
+#include <QSettings>
+#include <QSqlError>
+
+#include "library/queryutil.h"
+#include "library/trackcollectionmanager.h"
+#include "track/track.h"
+#include "util/performancetimer.h"
+
+namespace {
+constexpr int kClementineUndefinedBpm = -1;      // clementine saves undefined bpm as -1
+constexpr int kClementineUndefinedDuration = -1; // clementine saves undefined duration as -1
+} // namespace
+
+ClementineDbConnection::ClementineDbConnection()
+        : m_database() {
+}
+
+ClementineDbConnection::~ClementineDbConnection() {
+    qDebug() << "Close Clementine database";
+    m_database.close();
+}
+
+bool ClementineDbConnection::open(const QFileInfo& databaseFile) {
+    m_database = QSqlDatabase::addDatabase("QSQLITE", "CLEMENTINE_DB_CONNECTION");
+    m_database.setHostName("localhost");
+    m_database.setDatabaseName(databaseFile.filePath());
+    m_database.setConnectOptions("SQLITE_OPEN_READONLY");
+
+    //Open the database connection in this thread.
+    if (!m_database.open()) {
+        m_database.setConnectOptions(); // clear options
+        qWarning() << "Failed to open Clementine database." << m_database.lastError();
+        return false;
+    } else {
```

Comment:
Use early return, i.e. remove the redundant else arm and simply continue with the success path.